framework/src/media/MediaWorker.cpp

```diff
@@ -59,9 +59,40 @@ void MediaWorker::stopWorker()
 	}
 }
 
-MediaQueue& MediaWorker::getQueue()
+std::function<void()> MediaWorker::deQueue()
 {
-	return mWorkerQueue;
+	return mWorkerQueue.deQueue();
+}
+
+bool MediaWorker::idleCondition()
+{
+	return false;
+}
+
+void MediaWorker::idleWork()
+{
+}
+
+int MediaWorker::entry()
+{
+	medvdbg("MediaWorker::entry()\n");
+
+	while (mIsRunning) {
+		if (idleCondition()) {
```

Comment:
All these are excuted before dequeue,
how about modify as a below?
1. preProcessLoop//playback, record, do something before dequeue
2. processLoop//dequeue
3. postProcessLoop// after dequeue in future.

Then we can exapnd easily if there's chance that preemption needed or should do something right after dequeue.
Also rename entry to mediaLooper is better, I think

int MediaWorker::mediaLooper() 
{
      medvdbg("Looper started running state : %d\n", mIsRunning);
      
      while(mIsRunning) {
              if (preProcessLoop()) { 
                       continue;
              }
              if(processLoop()) {
                      postProcessLoop();
               } else {
                     meddbg("something wrong\n");
               }
       }
}
protected bool MediaWorker::preProcessLoop() {

      /** we need more looping because queue is empty */
      if (isEmpty()) {
           return true;
       }
}

protected bool RecordWorker::preProcessLoop() 
{
      if (mCurRecorder && (mCurRecorder->getState() == RECORDER_STATE_RECORDING)) {
               ret = capture();
       }
       if (ret) {
               return super;
       }
       return false;
}
```

Reply:
I agree with your idea, but it seems dangerous that derived worker must call the base worker method.
And in C++, super is not standard keyword